doc/xml/user-guide.xml

```diff
@@ -159,6 +159,13 @@
         <variable key="azure-key-type">shared</variable>
         <variable key="azure-key">YXpLZXk=</variable>
 
+        <!-- GCS Settings -->
+        <variable key="gcs-all">n</variable>                        <!-- Build all the documentation with GCS? -->
```

Comment:
So when I build the docs, I can build with azure `pgbackrest/doc/doc.pl --out=html --include=user-guide --var=encrypt=n --var=azure-all=y` and with s3 `pgbackrest/doc/doc.pl --out=html --include=user-guide --var=encrypt=n --var=s3-all=y`

But with I use `--var-gcs-all=y` I get:
```
P00   INFO: stanza-create for stanza 'demo' on repo1
P00  ERROR: [055]: unable to open missing file '/etc/pgbackrest/gcs-key.json' for read
```

Is there something I need to do first to get the json file?

Reply:
Yes -- you need a GCS account and to setup a service account (then download the key) and a bucket (with the correct permissions). Then you can add your key, .e.g. ` --var=gcs-key=/pgbackrest/test/scratch.gcs.json`.

On Azure we can run the docs locally on Azurite but FakeGCS is not up to the task.

Reply:
Actually, I was premature about s3-all - why is the stanza-delete using gcs when I've specified s3-all?
```
pgbackrest/doc/doc.pl --out=html --include=user-guide --var=encrypt=n --var=s3-all=y
2021-03-03 16:13:10.920 P00   INFO: source user-guide
2021-03-03 16:13:10.922 P00   INFO: Build vm 'pgbackrest/doc:debian' from 'ubuntu:16.04'
2021-03-03 16:13:11.540 P00   INFO: render html output
2021-03-03 16:13:11.612 P00   INFO:     render out: user-guide
2021-03-03 16:13:13.336 P00   INFO:         process section: /build
2021-03-03 16:13:13.369 P00   INFO:         process section: /installation
2021-03-03 16:13:13.436 P00   INFO:         process section: /quickstart
2021-03-03 16:13:13.440 P00   INFO:             process section: /quickstart/setup-demo-cluster
2021-03-03 16:13:13.479 P00   INFO:             process section: /quickstart/configure-stanza
2021-03-03 16:13:13.566 P00  ERROR: [028]: types do not match, cache is invalid
2021-03-03 16:13:13.568 P00   WARN: Cache will be reset for source user-guide and rendering retried automatically
2021-03-03 16:13:19.403 P00   INFO:         process section: /build
2021-03-03 16:14:47.461 P00   INFO:         process section: /installation
2021-03-03 16:14:56.877 P00   INFO:         process section: /quickstart
2021-03-03 16:14:56.883 P00   INFO:             process section: /quickstart/setup-demo-cluster
2021-03-03 16:15:00.907 P00   INFO:             process section: /quickstart/configure-stanza
2021-03-03 16:15:02.160 P00   INFO:             process section: /quickstart/create-repository
2021-03-03 16:15:03.865 P00   INFO:             process section: /quickstart/s3-support
2021-03-03 16:15:05.553 P00   INFO:             process section: /quickstart/configure-archiving
2021-03-03 16:15:14.159 P00   INFO:             process section: /quickstart/create-stanza
2021-03-03 16:15:15.061 P00   INFO:             process section: /quickstart/check-configuration
2021-03-03 16:15:16.738 P00   INFO:             process section: /quickstart/perform-backup
2021-03-03 16:15:29.160 P00   INFO:             process section: /quickstart/backup-info
2021-03-03 16:15:29.435 P00   INFO:             process section: /quickstart/perform-restore
2021-03-03 16:15:43.129 P00   INFO:         process section: /backup
2021-03-03 16:15:43.133 P00   INFO:             process section: /backup/option-start-fast
2021-03-03 16:15:50.310 P00   INFO:         process section: /monitor
2021-03-03 16:15:50.315 P00   INFO:             process section: /monitor/postgresql
2021-03-03 16:15:52.179 P00   INFO:             process section: /monitor/jq
2021-03-03 16:15:56.112 P00   INFO:         process section: /retention
2021-03-03 16:15:56.121 P00   INFO:             process section: /retention/full
2021-03-03 16:16:14.405 P00   INFO:             process section: /retention/diff
2021-03-03 16:16:25.194 P00   INFO:             process section: /retention/archive
2021-03-03 16:16:30.774 P00   INFO:         process section: /restore
2021-03-03 16:16:30.784 P00   INFO:             process section: /restore/option-delta
2021-03-03 16:16:37.883 P00   INFO:             process section: /restore/option-db-include
2021-03-03 16:16:55.536 P00   INFO:         process section: /pitr
2021-03-03 16:17:32.102 P00   INFO:         process section: /azure-support
2021-03-03 16:17:47.202 P00   INFO:         process section: /delete-stanza
2021-03-03 16:17:48.314 P00  ERROR: [125]: command 'docker exec -u vagrant doc-pg-primary bash -l -c 'sudo -u postgres pgbackrest --stanza=demo --log-level-console=info stanza-delete'' returned 55
                                        STDOUT (last 10,000 characters):
                                        P00   INFO: stanza-delete command begin 2.33dev: --exec-id=1768-96fed348 --log-level-console=info --log-level-stderr=off --no-log-timestamp --pg1-path=/var/lib/postgresql/12/demo --repo1-gcs-bucket=demo-bucket --repo1-gcs-key=/etc/pgbackrest/gcs-key.json --repo1-path=/demo-repo --repo1-type=gcs --stanza=demo
                                        P00  ERROR: [055]: unable to open missing file '/etc/pgbackrest/gcs-key.json' for read
                                        P00   INFO: stanza-delete command end: aborted with exception [055]
```

Reply:
Since the s3-all, azure-all and gcs-all are special testing modes for developing the particular driver and since multi-repo would make different drivers easier to work with (they would not be mixed in the doc generation) we have decided that this issue is not an issue.